inc/c_util/log_critical_and_terminate.h

```diff
@@ -0,0 +1,33 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+
+#ifndef LOG_CRITICAL_AND_TERMINATE_H
+#define LOG_CRITICAL_AND_TERMINATE_H
+
+#include "macro_utils/macro_utils.h"
+
+#include "c_logging/xlogging.h"
+
+#include "c_util/ps_util.h"
+
+#include "umock_c/umock_c_prod.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define LogCriticalAndTerminate(FORMAT, ...) \
+    do \
+    { \
+        static volatile int work_around_bs_utils_never_returning = 1; \
+        LogError(FORMAT, ##__VA_ARGS__); \
+        /* Codes_SRS_LOG_CRITICAL_AND_TERMINATE_01_001: [ LogCriticalAndTerminate shall call ps_util_terminate_process. ]*/ \
+        work_around_bs_utils_never_returning ? ps_util_terminate_process():(void)0; \
```

Comment:
>bs_utils [](start = 20, length = 8)

small nit: bs_utils leaked into c-util

Reply:
Fixed.

---
In reply to: [526530416](https://github.com/Azure/c-util/pull/80#discussion_r526530416) [](ancestors = 526530416)